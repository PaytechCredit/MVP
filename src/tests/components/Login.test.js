import React from "react";
// enzyme imports for smoke testing
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
// Redux Mock Store
import configureStore from "redux-mock-store";
// Testing Components
import Login from "pages/auth/LogIn";

const mockStore = configureStore();
const initialState = {
  auth: {
    isAuthenticated: false,
    loginError: {
      status: 0
    }
  }
};
const store = mockStore(initialState);

describe("Test case for testing login", () => {
  let wrapper;
  let testProps;

  beforeEach(() => {
    testProps = { store };
    wrapper = mount(<Login {...testProps} />, {
      disableLifecycleMethods: true
    });
  });

  // test("email check", () => {
  //   const email_input = wrapper.find('input[type="email"]');
  //   const instance = wrapper.instance();
  //   console.log(wrapper.props().store.getState("email"));
  //   email_input.simulate("change", {
  //     target: { name: "email", value: "johnsmith@ivylender.com" }
  //   });
  //   expect(wrapper.props()).toEqual("johnsmith@ivylender.com");
  // });
  test("renders the component", () => {
    const wrapper = shallow(<Login {...testProps} />);
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
  });
});
