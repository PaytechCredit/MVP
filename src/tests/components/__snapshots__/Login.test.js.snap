// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing login renders the component 1`] = `
<LogIn
  auth={
    Object {
      "isAuthenticated": false,
      "loginError": Object {
        "status": 0,
      },
    }
  }
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
